sql/src/test/java/io/crate/analyze/PrivilegesDCLAnalyzerTest.java

```diff
@@ -135,22 +227,51 @@ public void testGrantWithoutUserManagementEnabledThrowsException() {
     }
 
     @Test
-    public void testRevokeWithoutUserManagementEnabledThrowsException() {
-        expectedException.expect(UnsupportedOperationException.class);
-        expectedException.expectMessage("User management is not enabled");
-        e = SQLExecutor.builder(clusterService).build();
-        e.analyze("REVOKE DQL FROM test");
+    public void testGrantOnUnknownSchemaThrowsException() {
+        expectedException.expect(SchemaUnknownException.class);
+        expectedException.expectMessage("Schema 'hoichi' unknown");
+        analyzePrivilegesStatement("GRANT DQL on schema hoichi TO user1");
+    }
+
+    @Test
+    public void testGrantOnUnknownTabkeThrowsException() {
+        expectedException.expect(TableUnknownException.class);
+        expectedException.expectMessage("Table 'doc.hoichi' unknown");
+        analyzePrivilegesStatement("GRANT DQL on table doc.hoichi TO user1");
+    }
+
+    @Test
+    public void testRevokeFromUnknownSchemaThrowsException() {
+        expectedException.expect(SchemaUnknownException.class);
+        expectedException.expectMessage("Schema 'hoichi' unknown");
+        analyzePrivilegesStatement("Revoke DQL on schema hoichi FROM user1");
+    }
+
+    @Test
+    public void testRevokeFromUnknownTableThrowsException() {
+        expectedException.expect(TableUnknownException.class);
+        expectedException.expectMessage("Table 'doc.hoichi' unknown");
+        analyzePrivilegesStatement("Revoke DQL on table doc.hoichi FROM user1");
     }
 
     private Privilege privilegeOf(Privilege.State state, Privilege.Type type) {
         return new Privilege(state,
```

Comment:
pls call the new privilegeOf(Privilege.State state, Privilege.Type type, Privilege.Clazz clazz, String ident) method instead